code/game/machinery/cryopod.dm

```diff
@@ -0,0 +1,418 @@
+
+// This is not a good situation
+/obj/machinery/cryopod/Destroy()
+	control_computer = null
+	return ..()
+
+/obj/machinery/cryopod/proc/find_control_computer(urgent = FALSE)
+
+	for(var/cryo_console as anything in GLOB.cryopod_computers)
+		var/obj/machinery/computer/cryopod/console = cryo_console
+		if(get_area(console) == get_area(src))
+			control_computer = console
+			break
+
+	// Don't send messages unless we *need* the computer, and less than five minutes have passed since last time we messaged
+	if(!control_computer && urgent && COOLDOWN_FINISHED(src, last_no_computer_message))
+		COOLDOWN_START(src, last_no_computer_message, 5 MINUTES)
+		log_admin("Cryopod in [get_area(src)] could not find control computer!")
+		message_admins("Cryopod in [get_area(src)] could not find control computer!")
+		last_no_computer_message = world.time
+
+	return control_computer != null
+
+/obj/machinery/cryopod/close_machine(mob/user)
+	if(!control_computer)
+		find_control_computer(TRUE)
+	if((isnull(user) || istype(user)) && state_open && !panel_open)
+		..(user)
+		var/mob/living/mob_occupant = occupant
+		if(mob_occupant && mob_occupant.stat != DEAD)
+			to_chat(occupant, "<span class='boldnotice'>You feel cool air surround you. You go numb as your senses turn inward.</span>")
+		if(mob_occupant.client) // if they're logged in
+			despawn_world_time = world.time + (time_till_despawn * 0.1) // This gives them 30 seconds
+		else
+			despawn_world_time = world.time + time_till_despawn
+	icon_state = "cryopod"
+
+/obj/machinery/cryopod/open_machine()
+	..()
+	icon_state = "cryopod-open"
+	density = TRUE
+	name = initial(name)
+
+/obj/machinery/cryopod/container_resist_act(mob/living/user)
+	visible_message("<span class='notice'>[occupant] emerges from [src]!</span>",
+		"<span class='notice'>You climb out of [src]!</span>")
+	open_machine()
+
+/obj/machinery/cryopod/relaymove(mob/user)
+	container_resist_act(user)
+
+/obj/machinery/cryopod/process()
+	if(!occupant)
+		return
+
+	var/mob/living/mob_occupant = occupant
+	if(mob_occupant)
+		// Eject dead people
+		if(mob_occupant.stat == DEAD)
+			open_machine()
+
+		if(!(world.time > despawn_world_time))
+			return
+
+		if(!mob_occupant.client && mob_occupant.stat < 2) // Occupant is living and has no client.
+			if(!control_computer)
+				find_control_computer(urgent = TRUE) // better hope you found it this time
+
+			despawn_occupant()
+
+/obj/machinery/cryopod/proc/handle_objectives()
+	var/mob/living/mob_occupant = occupant
+	// Update any existing objectives involving this mob.
+	for(var/datum/objective/objective in GLOB.objectives)
+		// We don't want revs to get objectives that aren't for heads of staff. Letting
+		// them win or lose based on cryo is silly so we remove the objective.
+		if(istype(objective,/datum/objective/mutiny) && objective.target == mob_occupant.mind)
+			objective.team.objectives -= objective
+			qdel(objective)
+			for(var/datum/mind/M in objective.team.members)
```

Comment:
Fix ALL the names. I'm not bothering to point them all out specifically.

https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names